consensus/merge/src/test/java/org/hyperledger/besu/consensus/merge/TransitionProtocolScheduleTest.java

```diff
@@ -195,16 +179,64 @@ public void getByBlockNumber_delegatesToPostMergeScheduleWhenTimestampScheduleDo
 
     transitionProtocolSchedule.getByBlockNumber(BLOCK_NUMBER);
 
-    verifyPostMergeProtocolScheduleReturned();
+    verifyPostMergeProtocolScheduleReturnedUsingBlockNumber();
+  }
+
+  @Test
+  public void getByBlockHeader_returnsTimestampScheduleIfPresent() {
+    when(timestampSchedule.getByTimestamp(TIMESTAMP))
+        .thenReturn(Optional.of(mock(ProtocolSpec.class)));
+
+    assertThat(transitionProtocolSchedule.getByBlockHeader(blockHeader)).isNotNull();
+
+    verify(timestampSchedule).getByTimestamp(TIMESTAMP);
+    verifyNoMergeScheduleInteractions();
+  }
+
+  @Test
+  public void getByBlockHeader_delegatesToPreMergeScheduleWhenBlockNotFound() {
+    when(mergeContext.isPostMerge()).thenReturn(false);
+
+    transitionProtocolSchedule.getByBlockHeader(blockHeader);
+
+    verifyPreMergeProtocolScheduleReturnedUsingBlockHeader();
+  }
+
+  @Test
+  public void getByBlockHeader_delegatesToPostMergeScheduleWhenBlockNotFound() {
```

Comment:
How is this distinguished from the test below, `getByBlockHeader_delegatesToPostMergeScheduleWhenTimestampScheduleDoesNotExist`?

Before, it was testing 'block not found' with this...
`when(blockchain.getBlockHeader(BLOCK_NUMBER)).thenReturn(Optional.empty());`

Reply:
Fixed the tests for the block not found cases for `getByBlockNumber`. And have removed tests for `getByBlockHeader` for the block not found cases. For `getByBlockHeader` we don't need to do a `blockchain.getBlockHeader()` call so we can never get the case where the block or block header doesn't exist.